modules/budget/variables.tf

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "billing_account" {
+  description = "ID of the billing account to set a budget on"
+  type        = string
+}
+
+variable "projects" {
+  description = "The project ids to include in this budget. If empty budget will include all projects"
+  type        = list(string)
+}
+
+variable "amount" {
+  description = "The amount to use as the budget"
+  type        = string
```

Comment:
Should this be a number?

Reply:
Provider resource is a `string`. 
https://www.terraform.io/docs/providers/google/r/billing_budget.html#units

Maybe because it can be something other than USD

Reply:
Weird, could we be more restrictive and mandate a number (which can be safely coerced to a string)?

Reply:
Made it a `number` with a `tostring` call in the module